qiskit/synthesis/discrete_basis/commutator_decompose.py

```diff
@@ -89,11 +89,11 @@ def _solve_decomposition_angle(matrix: np.ndarray) -> float:
     trace = _compute_trace_so3(matrix)
     angle = math.acos((1 / 2) * (trace - 1))
 
+    lhs = math.sin(angle / 2)
+
     def objective(phi):
-        rhs = 2 * math.sin(phi / 2) ** 2
-        rhs *= math.sqrt(1 - math.sin(phi / 2) ** 4)
-        lhs = math.sin(angle / 2)
-        return rhs - lhs
+        sin_sq = np.sin(phi / 2) ** 2
+        return 2 * sin_sq * np.sqrt(1 - sin_sq**2) - lhs
```

Comment:
I assume this was the place with the implicit scalar conversion was emitting a deprecation warning because of the switch from `math.sin()` and `math.sqrt()` to `np.sin()` and np.sqrt()`.

Reply:
This is one of them, yeah.  I didn't realise I'd screwed with the implementation as much as that - I think I was thinking about reducing the increased overhead from switching back to Numpy dispatchers.  I don't know enough about the internals of how Scipy's `fsolve` to know for sure what's safe as a return value, but it definitely puts in arrays.